src/actions/ast.js

```diff
@@ -126,3 +138,64 @@ export function setSelection(
     });
   };
 }
+
+/**
+ * Action dispatched from stepOver action. Receives the position
+ * and depending on where it is paused, sets up hidden breakpoints
+ * and handles resume, stepOver (as needed)
+ * @param position
+ * @returns {function(ThunkArgs)}
+ */
+export function analyzeStepping(type) {
+  return async ({ dispatch, getState, client }: ThunkArgs) => {
+    const position = await dispatch(getPosition());
+    const source = getSelectedSource(getState()).toJS();
+    const hiddenBreakpointLocation = getHiddenBreakpoint(getState());
+
+    await dispatch({
+      type: "ANALYZE_STEPPING",
+      [PROMISE]: (async function() {
+        if (hiddenBreakpointLocation) {
+          await dispatch(removeBreakpoint(hiddenBreakpointLocation));
+        }
+        const path = getClosestPath(source, position);
+        if (!path) {
+          return dispatch(command({ type }));
+        }
+        switch (type) {
```

Comment:
I personally prefer the style `["stepOver","stepIn",].includes(type)` or `type == "stepOver" || type == "stepOver",`